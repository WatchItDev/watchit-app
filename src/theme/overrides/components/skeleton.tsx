import { Theme } from '@mui/material/styles';

// ----------------------------------------------------------------------

export function skeleton(theme: Theme) {
  return {
    MuiSkeleton: {
      styleOverrides: {
        root: {
          backgroundColor: theme.palette.background.neutral,
        },
        rounded: {
          borderRadius: theme.shape.borderRadius * 2,
        },
      },
    },
  };
}
