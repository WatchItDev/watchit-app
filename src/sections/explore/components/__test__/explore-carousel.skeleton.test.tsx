import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import { ExploreCarouselSkeleton } from "../explore-carousel.skeleton";


vi.mock('@src/hooks/components/use-item-per-slide', () => ({
  useItemsPerSlide: () => ({
    itemsPerSlide: 3,
  }),
}));

describe("Testing in the <ExploreCarouselSkeleton/> component", () => {
  it("should match snapshot", () => {
    const { container } = render(<ExploreCarouselSkeleton title="Test" />);
    expect(container).toMatchSnapshot();
  });

  it("should render the title", () => {
    const { getByText } = render(<ExploreCarouselSkeleton title="Test" />);
    expect(getByText("Test")).toBeInTheDocument();
  });

  it('should render 6 skeleton items if itemsPerSlide is 3', () => {
    const { getAllByTestId } = render(<ExploreCarouselSkeleton title="Test" />);
    const skeletonItems = getAllByTestId('skeleton-item');
    expect(skeletonItems.length).toBe(6);
  });
});
