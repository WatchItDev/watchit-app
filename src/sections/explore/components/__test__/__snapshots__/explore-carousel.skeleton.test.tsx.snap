// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing in the <ExploreCarouselSkeleton/> component > should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-19i2hl5-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-hl0h5w-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
        >
          Test
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-ceyv38-MuiCardContent-root"
    >
      <div
        class="MuiBox-root css-i6bazn"
      >
        <div
          class="MuiBox-root css-k008qs"
        >
          <div
            class="MuiBox-root css-b3j2rt"
            data-testid="skeleton-item"
          >
            <div
              class="MuiBox-root css-0"
            />
          </div>
          <div
            class="MuiBox-root css-b3j2rt"
            data-testid="skeleton-item"
          >
            <div
              class="MuiBox-root css-0"
            />
          </div>
          <div
            class="MuiBox-root css-b3j2rt"
            data-testid="skeleton-item"
          >
            <div
              class="MuiBox-root css-0"
            />
          </div>
        </div>
        <div
          class="MuiBox-root css-k008qs"
        >
          <div
            class="MuiBox-root css-b3j2rt"
            data-testid="skeleton-item"
          >
            <div
              class="MuiBox-root css-0"
            />
          </div>
          <div
            class="MuiBox-root css-b3j2rt"
            data-testid="skeleton-item"
          >
            <div
              class="MuiBox-root css-0"
            />
          </div>
          <div
            class="MuiBox-root css-b3j2rt"
            data-testid="skeleton-item"
          >
            <div
              class="MuiBox-root css-0"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
